services/requester/batch_tx_pool.go

```diff
@@ -22,35 +23,42 @@ import (
 	"github.com/onflow/flow-evm-gateway/services/requester/keystore"
 )
 
-const eoaActivityCacheSize = 10_000
+const (
+	eoaActivityCacheSize     = 10_000
+	maxTrackedTxNoncesPerEOA = 15
```

Comment:
is 15 enough? I think the data we'd need to cache is small enough we could use a larger value here without tying up too much memory.